### ISSUE TITLE
The link interface of target "exiv2lib" contains: ZLIB::ZLIB but the target was not found

### ISSUE THREAD

**Issue by Ryanf55** [Exiv2/exiv2#2687]
#### **Describe the bug**

Exiv2 is build with the following flags:
```
cmake -B build -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=Off -DEXIV2_ENABLE_XMP=Off -DEXIV2_ENABLE_INIH=Off
cmake --build build --parallel `nproc`
sudo cmake --install build
```

Then, it's consumed with `find_package(exiv2 REQUIRED)`. 

It's missing a call to `find_dependency` in the export. 
https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html#command:find_dependency

The generated cmake config file has the following code:
```cmake
set_target_properties(exiv2lib PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:ZLIB::ZLIB>;/usr/lib/x86_64-linux-gnu/libbrotlicommon.so;/usr/lib/x86_64-linux-gnu/libbrotlidec.so"
)
```
It's linking to `ZLIB` but that target doesn't exist because it was never requested to be found. 

#### **Expected behavior**

If zlib is required, it's added as an export dependency.

#### **Desktop (please complete the following information):**
- OS and version: Ubuntu 22.04
- Exiv2 version and source:  Branch:  v0.28.0 hash 931a40a746f5678dcc4625b06a2eb25fa4f00b34
- Compiler and version: GCC11.3.0
- CMake version: 3.27

### Steps to reproduce
```bash
cd /path/to/exiv2
docker run -v $(pwd):/ws -w /ws --net=host -it ubuntu:22.04 bash
apt update && apt install -y build-essential cmake libbrotli-dev libz-dev
cmake -B build -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=Off -DEXIV2_ENABLE_XMP=Off -DEXIV2_ENABLE_INIH=Off
cmake --build build --parallel `nproc`
cmake --install build
cat /usr/local/lib/cmake/exiv2/exiv2Config.cmake
```

Now, in another out-of-tree directory like `/project/` create a  CMakeLists that links to exiv2

**CMakeLists.txt**
```cmake
cmake_minimum_required(VERSION 3.22)
project(exiv2_zlib_install)

find_package(exiv2 CONFIG REQUIRED)

add_executable(main main.cpp)
target_link_libraries(main PUBLIC exiv2lib)
```
**main.cpp**
```c++
int main() {
        return 0;
}
```
Finally, build the little example.
```bash
cd /project
cmake -B build
...
-- Configuring done
CMake Error at CMakeLists.txt:6 (add_executable):
  Target "main" links to target "ZLIB::ZLIB" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
```

**Comment by neheb** [comment]
I don't speak CMake. Patch maybe?

**Comment by Ryanf55** [comment]
> I don't speak CMake. Patch maybe?

I can try. Can you assign this issue to me?

**Comment by Ryanf55** [comment]
Closed by #2703 .

Other libraries will face the same problem that are part of the link interface, however the pattern is now established how to fix it.
